signinapp/forms.py

```diff
@@ -0,0 +1,75 @@
+from datetime import datetime
+
+from flask_wtf import FlaskForm
+from wtforms import (
+    BooleanField,
+    EmailField,
+    Field,
+    Form,
+    FormField,
+    IntegerField,
+    PasswordField,
+    SelectField,
+    StringField,
+    SubmitField,
+    TelField,
+)
+from wtforms.validators import DataRequired
+
+from .model import ShirtSizes
+
+
+class DataMaybeRequired(DataRequired):
+    def __init__(self, message=None):
+        super().__init__(message)
+
+    def __call__(self, form, field: Field):
+        if field.flags.is_required:
+            return super().__call__(form, field)
+
+
+class StudentDataForm(Form):
+    graduation_year = IntegerField(
```

Comment:
This should be a `SelectField` with `generate_grade_choices()` for the choices. We can deal with alumni in the future.